### PR TITLE
TMA-1003: Add unit test for delete_extra_process_schedule

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/911/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1724/ : SUCCESS in 2m 52s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1985/ : SUCCESS in 2m 38s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/162/ : SUCCESS in 6m 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/912/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1725/ : SUCCESS in 2m 54s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1986/ : SUCCESS in 1m 25s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/163/ : SUCCESS in 12m 06s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/913/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1726/ : SUCCESS in 1m 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1987/ : SUCCESS in 1m 14s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/164/ : SUCCESS in 8m 25s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/914/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1727/ : SUCCESS in 1m 26s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1988/ : SUCCESS in 3m 14s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/165/ : SUCCESS in 9m 58s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: spec/unit/actions/synchronize_etls_in_segment_spec.rb [hunk 1]
```diff
@@ -30,20 +47,97 @@
...
-    expect(project).to receive(:set_metadata)
+    expect(target_project).to receive(:set_metadata)
       .with('GOODOT_CUSTOM_PROJECT_ID', client_id)
     subject.class.call(params)
   end
 
+  context 'when delete_extra_process_schedule is true' do
+    let(:custom_params) { params.merge(delete_extra_process_schedule: false) }
+
+    it 'keeps all processes in target project' do
```
Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: warning: Similar code found in synchronize_etls_in_segment_spec.rb:93, synchronize_etls_in_segment_spec.rb:99 (mass = 38) (flay)

File: spec/unit/actions/synchronize_etls_in_segment_spec.rb [hunk 2]
```diff
@@ -30,20 +47,97 @@
...
+      expect(target_process).not_to receive(:delete)
+      expect(custom_target_process).not_to receive(:delete)
+      GoodData::LCM2.run_action(described_class, custom_params)
+    end
+
+    it 'keeps all schedules in target project' do
```
Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: warning: Similar code found in synchronize_etls_in_segment_spec.rb:93, synchronize_etls_in_segment_spec.rb:99 (mass = 38) (flay)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/916/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1729/ : FAILURE in 1m 13s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1990/ : SUCCESS in 1m 15s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/167/ : SUCCESS in 7m 20s

File: spec/unit/actions/synchronize_etls_in_segment_spec.rb [hunk 3]
```diff
@@ -7,21 +7,52 @@
 require 'gooddata/lcm/lcm2'
 require 'gooddata/lcm/actions/synchronize_etls_in_segment'
 
+def process_mock(name, id)
```
Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: warning: Similar code found in synchronize_etls_in_segment_spec.rb:10, synchronize_etls_in_segment_spec.rb:17 (mass = 42) (flay)

Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: info: Has approx 6 statements (TooManyStatements) (reek)

File: spec/unit/actions/synchronize_etls_in_segment_spec.rb [hunk 4]
```diff
@@ -7,21 +7,52 @@
...
+  mock = double(GoodData::Process)
+  allow(mock).to receive(:name) { name }
+  allow(mock).to receive(:process_id) { id }
+  mock
+end
+
+def schedule_mock(name, process_id)
```
Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: warning: Similar code found in synchronize_etls_in_segment_spec.rb:10, synchronize_etls_in_segment_spec.rb:17 (mass = 42) (flay)

Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: info: Has approx 6 statements (TooManyStatements) (reek)

File: spec/unit/actions/synchronize_etls_in_segment_spec.rb [hunk 5]
```diff
@@ -30,20 +61,88 @@
...
-    expect(project).to receive(:set_metadata)
+    expect(target_project).to receive(:set_metadata)
       .with('GOODOT_CUSTOM_PROJECT_ID', client_id)
     subject.class.call(params)
   end
 
+  context 'when delete_extra_process_schedule is true' do
+    let(:custom_params) { params.merge(delete_extra_process_schedule: false) }
+
+    it 'keeps all processes in target project' do
```
Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: warning: Similar code found in synchronize_etls_in_segment_spec.rb:98, synchronize_etls_in_segment_spec.rb:104 (mass = 38) (flay)

File: spec/unit/actions/synchronize_etls_in_segment_spec.rb [hunk 6]
```diff
@@ -30,20 +61,88 @@
...
+      expect(target_process).not_to receive(:delete)
+      expect(custom_target_process).not_to receive(:delete)
+      GoodData::LCM2.run_action(described_class, custom_params)
+    end
+
+    it 'keeps all schedules in target project' do
```
Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: warning: Similar code found in synchronize_etls_in_segment_spec.rb:98, synchronize_etls_in_segment_spec.rb:104 (mass = 38) (flay)

File: spec/unit/actions/synchronize_etls_in_segment_spec.rb [hunk 7]
```diff
@@ -7,21 +7,52 @@
 require 'gooddata/lcm/lcm2'
 require 'gooddata/lcm/actions/synchronize_etls_in_segment'
 
+def process_mock(name, id)
+  mock = double(GoodData::Process)
+  allow(mock).to receive(:name) { name }
```
Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: info: Calls 'allow(mock)' 2 times (DuplicateMethodCall) (reek)

File: spec/unit/actions/synchronize_etls_in_segment_spec.rb [hunk 8]
```diff
@@ -7,21 +7,52 @@
...
+  allow(mock).to receive(:process_id) { id }
+  mock
+end
+
+def schedule_mock(name, process_id)
+  mock = double(GoodData::Schedule)
+  allow(mock).to receive(:name) { name }
```
Comment (yenkins) [inline spec/unit/actions/synchronize_etls_in_segment_spec.rb]: info: Calls 'allow(mock)' 2 times (DuplicateMethodCall) (reek)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/927/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1740/ : FAILURE in 1m 25s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/2001/ : SUCCESS in 1m 21s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/178/ : SUCCESS in 8m 53s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/157/ : SUCCESS in 34s

**Pull request merged**